transcom/mymove
# B-21362 & B-212363 UB Shipment DB/Model Updates/Customer API - INT

## B-21362

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21362)

## Summary

We are adding a new shipment type, UNACCOMPANIED_BAGGAGE, which is the same as an HHG shipment in structure. In coordination with Beth as the DB SME, we decided to not have a new table created for the UB shipment. Instead, we will utilize the existing mto_shipments table, with an expanded shipment_type enum value of UNACCOMPANIED_BAGGAGE.

Work in this PR:
- A new feature flag was added for UB shipments
milmove-feature-flags repo PR: https://github.com/transcom/milmove-feature-flags/pull/209
[add UNACCOMPANIED_BAGGAGE to envrc and circle configs](https://github.com/transcom/mymove/commit/62cd9bd1a1aefd747d7cc8f8dbc973423786d248)
[add unaccompanied_baggage to config/flipt](https://github.com/transcom/mymove/commit/ede4db21bd9d0fad9be7d21c9812e72336d8a760)

- The mto_shipment_type was updated to include the UNACCOMPANIED_BAGGAGE type
- The mto_shipment_type was updated to exclude the unused types of INTERNATIONAL_HHG and INTERNATIONAL_UB. Instead of those types, we will have a market_code of 'i' or 'd' on the shipment to designate HHGs and UBs as international or not
[migration for ub shipment type, removing unused intl-hhg and intl-ub](https://github.com/transcom/mymove/commit/45576b6685850873d327f4782fdd083eb9c65702)

- Updated the MTO shipment model and swagger to include the new UB type
[add ub to swagger and model](https://github.com/transcom/mymove/commit/c0a688aa53b0b7144f8d809e3dfecb5241edc512)

- Updated support API swagger def to have all current shipment types, were missing a few
[update support swagger def to have all shipment types](https://github.com/transcom/mymove/commit/5c3861e4753a8d9d7ae2364fcab3cb326ec5b431)

- Removed existing code references to the removed international HHG and international UB shipment types, made sure tests still passed
[remove refs/tests using intl hhg and intl ub, tests passing](https://github.com/transcom/mymove/commit/2943d5eb2a0de4c60cc7afafe85f77b6675b643d)

To test:
1. Run the migration and verify that you see the UNACCOMPANIED_BAGGAGE as an available shipment type to select.
![image](https://github.com/user-attachments/assets/227f8061-71b9-4ae3-a3c4-06e5cd4027c2)
__________________________________________________________________________________________
## B-21363

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21363)

## Summary

B-21362 specific commits:
- [add ub check for populating dest address in shipment creator](https://github.com/transcom/mymove/pull/13837/commits/f49d7a996d1df301a9cec9eec57d572493002c58)
[add UB chcek to eval report form creator](https://github.com/transcom/mymove/pull/13837/commits/45728053f6a53d0ae46f1b5f8177a50217ead2b7)

With the addition of the new shipment type, UNACCOMPANIED_BAGGAGE, which is the same as an HHG shipment in structure, we need to make sure that the customer API endpoints are working as expected to create, update, and fetch UB shipments.

To test:
1. As a customer, create a move and add an HHG shipment. Don't finish out the move, and park yourself on the page that shows you your created shipments.
2. Test out createMTOShipment `http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/createMTOShipment`, making sure you get a 200 response and that a new shipment with the UNACCOMPANIED BAGGAGE type exists in the database. Example body:
```
{
  "moveTaskOrderID": "91154a53-6880-460b-a270-57bcb53d631a",
  "shipmentType": "UNACCOMPANIED_BAGGAGE",
  "requestedPickupDate": "2024-10-04",
  "pickupAddress": {
      "streetAddress1": "123 Main St",
      "city": "Nicholasville",
      "state": "KY",
      "postalCode": "40356",
      "country": "USA",
      "county": "JESSAMINE"
   }
}
```
3. Create another one if you want. Check the UI and make sure a shipment shows up. It won't have any labels yet and might look weird. Check the DB again to make sure you now have two UB shipments if you did create another.
![image](https://github.com/user-attachments/assets/b963d284-6a9c-41b6-8cde-79f07bc76683)
![image](https://github.com/user-attachments/assets/a8804d4e-9fe6-4aee-a70d-4fc1d4b7ded6)

4. Now test out updating the shipment. Make sure you get a 200 for `http://milmovelocal:3000/swagger-ui/internal.html#/mtoShipment/updateMTOShipment`. Example body where I changed the zip code and state:
```
{
  "moveTaskOrderID": "91154a53-6880-460b-a270-57bcb53d631a",
  "shipmentType": "UNACCOMPANIED_BAGGAGE",
  "requestedPickupDate": "2024-10-04",
  "pickupAddress": {
      "streetAddress1": "123 Main St",
      "city": "Nicholasville",
      "state": "KS",
      "postalCode": "40358",
      "country": "USA",
      "county": "JESSAMINE"
   }
}
```
5. Make sure whatever you changed with the body is now reflected in the DB.

6. Test that you see all shipments, especially the UBs, listed out and get a 200 for listMTOShipments with `http://milmovelocal:3000/internal/moves/91154a53-6880-460b-a270-57bcb53d631a/mto_shipments`.

7. Test that you see the new UB shipments in the response for getAllMoves for the service member with `http://milmovelocal:3000/swagger-ui/internal.html#/moves/getAllMoves`
